index.js

```diff
@@ -1,16 +1,36 @@
 'use strict';
 module.exports = function (hex) {
-	if (typeof hex !== 'string') {
-		throw new TypeError('Expected a string');
+	if (
+		typeof hex !== 'string' ||
+		/[^#a-f0-9]/gi.test(hex) ||
+		!/^#?[a-f0-9]{3}[a-f0-9]?$|^#?[a-f0-9]{6}([a-f0-9]{2})?$/i.test(hex)
```

Comment:
For readability reasons, can you make these two regexes functions with descriptive names?

Reply:
And `0-9` => `\d`